src/main/scala/org/ergoplatform/nodeView/mempool/OrderedTxPool.scala

```diff
@@ -152,17 +182,12 @@ object OrderedTxPool {
   private implicit val ordBoxId: Ordering[BoxId] = Ordering[String].on(b => Algos.encode(b))
 
   def empty(settings: ErgoSettings): OrderedTxPool = {
-    OrderedTxPool(TreeMap.empty[WeightedTxId, ErgoTransaction], TreeMap.empty[ModifierId, WeightedTxId],
-      TreeMap.empty[ModifierId, Long], TreeMap.empty[BoxId, WeightedTxId])(settings)
-  }
-
-  private def weighted(tx: ErgoTransaction)(implicit ms: MonetarySettings): WeightedTxId = {
-    val fee = tx.outputs
-      .filter(b => java.util.Arrays.equals(b.propositionBytes, ms.feePropositionBytes))
-      .map(_.value)
-      .sum
-    // We multiply by 1024 for better precision
-    WeightedTxId(tx.id, fee * 1024 / tx.size)
+    OrderedTxPool(
+      TreeMap.empty[WeightedTxId, ErgoTransaction],
+      TreeMap.empty[ModifierId, WeightedTxId],
+      TreeMap.empty[ModifierId, Long],
+      TreeMap.empty[BoxId, WeightedTxId],
+      TreeMap.empty[BoxId, WeightedTxId])(settings)
```

Comment:
Since these TreeMaps is immutable, to avoid garbage, empty instances can be allocated once and saved as `val`, then shared by all new OrderedTxPool created here.

Reply:
Empty pool is created only once, so not an issue likely.